adam/curriculum/pursuit_curriculum.py

```diff
@@ -0,0 +1,118 @@
+    Args:
+        num_instances: total number of learning instances for each object to learn
+        num_noise_instances: number of noisy learning instances in each set of learning instances. A noisy instance in
+        a scene where the utterance doesn't match the situation and perception (e.g. hearing "ball" while seeing a cup).
+        num_objects_in_instance: number of objects in each instance
+    """
+    if num_noise_instances > num_instances:
+        raise RuntimeError("Cannot have more noise than regular exemplars")
+
+    target_objects = [BALL, CHAIR, PERSON, TABLE, DOG, BIRD, BOX]
+    target_object_variables = [
+        object_variable(target.handle + "-target", target) for target in target_objects
+    ]
+    other_objects = [
+        standard_object("obj-" + str(idx)) for idx in range(num_objects_in_instance)
+    ]
+
+    # A list of templates with specific a target object in each to create learning instances. Specifically,
+    # there is one object (e.g. Ball) across all instances while the other objects vary. Hence, the target object is
+    # a salient object (used for generating a dependency tree) while the remaining objects are background objects.
+    object_is_present_templates = [
+        Phase1SituationTemplate(
+            "simple_pursuit",
+            salient_object_variables=[target_object_variable],
+            background_object_variables=other_objects[:-1],
+        )
+        for target_object_variable in target_object_variables
+    ]
+
+    # A generic template that is used to replace situations and perceptions (not dependency trees) in noise instances
+    noise_template = Phase1SituationTemplate(
+        "simple_pursuit",
+        salient_object_variables=[other_objects[0]],
+        background_object_variables=other_objects[1:],
+    )
+
+    all_instances = []
+    # Generate phase_1 instance groups for each template (i.e each target word)
+    for object_is_present_template in object_is_present_templates:
+        new_instances = list(
+            phase1_instances(
+                "simple_pursuit_curriculum",
+                sampled(
+                    object_is_present_template,
+                    max_to_sample=num_instances - num_noise_instances,
+                    chooser=PHASE1_CHOOSER,
+                    ontology=GAILA_PHASE_1_ONTOLOGY,
+                ),
+            ).instances()
+        )
+        all_instances.extend(new_instances)
+        # Create instances for noise
+        if num_noise_instances > 0:
+            noise_instances = phase1_instances(
+                "simple_pursuit_curriculum",
+                sampled(
+                    noise_template,
+                    max_to_sample=num_noise_instances,
+                    chooser=PHASE1_CHOOSER,
+                    ontology=GAILA_PHASE_1_ONTOLOGY,
+                ),
+            ).instances()
+            for noise_instance in noise_instances:
```

Comment:
Python's tuple de-structuring can make this clearer:

```
for (situation, linguistic_description, perception) in noise_instances:
    ...
```

Also comment in this section describing why this is necessary.